apps/heft/src/cli/actions/DevDeployAction.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { HeftActionBase, IHeftActionBaseOptions, ActionHooksBase, IActionDataBase } from './HeftActionBase';
+
+/**
+ * @public
+ */
+export interface IDevDeployActionData extends IActionDataBase<DevDeployHooks> {}
+
+/**
+ * @public
+ */
+export class DevDeployHooks extends ActionHooksBase {}
+
+export class DevDeployAction extends HeftActionBase<IDevDeployActionData, DevDeployHooks> {
+  public constructor(options: IHeftActionBaseOptions) {
+    super(
+      {
+        actionName: 'dev-deploy',
+        summary: 'Deploy the current project, and optionally the whole repo, to a testing CDN.',
```

Comment:
How will the out-of-box implementation of this work? Is this operation generic enough across repos for it to be one of the default built-in actions?

Reply:
Right now it's been designed to work with Azure storage, with hooks for asset discovery and processing and account and storage key. It could easily be further genericized.